components/plate_based_widgets/mapeditor/PlateMapEditor.vue

```diff
@@ -1,725 +1,335 @@
-          } else {
-            new_list[4] = new_list[5] = new_list[6] = new_list[7] = true;
-          }
-          break;
-        case "3":
-          result = new_list[8] && new_list[9] && new_list[10] && new_list[11];
-          if (result == true) {
-            new_list[8] = new_list[9] = new_list[10] = new_list[11] = false;
-          } else {
-            new_list[8] = new_list[9] = new_list[10] = new_list[11] = true;
-          }
-          break;
-        case "4":
-          result = new_list[12] && new_list[13] && new_list[14] && new_list[15];
-          if (result == true) {
-            new_list[12] = new_list[13] = new_list[14] = new_list[15] = false;
-          } else {
-            new_list[12] = new_list[13] = new_list[14] = new_list[15] = true;
-          }
-          break;
-        case "5":
-          result = new_list[16] && new_list[17] && new_list[18] && new_list[19];
-          if (result == true) {
-            new_list[16] = new_list[17] = new_list[18] = new_list[19] = false;
-          } else {
-            new_list[16] = new_list[17] = new_list[18] = new_list[19] = true;
-          }
-          break;
-        case "6":
-          result = new_list[20] && new_list[21] && new_list[22] && new_list[23];
-          if (result == true) {
-            new_list[20] = new_list[21] = new_list[22] = new_list[23] = false;
-          } else {
-            new_list[20] = new_list[21] = new_list[22] = new_list[23] = true;
-          }
-          break;
-      }
-
-      this.all_select = new_list;
-      const allEqual = (arr) => arr.every((v) => v === true); // verify in the pre-select all via a const allEqual function.
-      this.all_select_or_cancel = allEqual(this.all_select) ? false : true; // if pre-select has all wells is true, then toggle from (+) to (-) icon.
-
-      for (let i = 0; i < this.all_select.length; i++) {
-        this.stroke_width[i] = !this.all_select[i] ? no_stroke_width : selected_stroke_width;
-        this.hover_color[i] = !this.all_select[i] ? hover_color : selected_color;
+    handle_btn_click: function ({ target }) {
+      const button_idx = Number(target.id);
+      if (button_idx === 0 && this.invalid_text === "") {
+        // saving name on save instead of here as it's input to trigger dropdown change in other component
+        this.save_platemap(this.input_platemap_name);
+      } else if (button_idx === 1) {
+        this.clear_all_well_assignments();
+        this.input_platemap_name = "";
+      } else if (button_idx === 2) {
+        this.discard_current_platemap_changes();
+        this.input_platemap_name = this.current_platemap_name;
       }
-      this.on_plate_well_selected();
     },
-    on_column_enter_hover(value) {
-      this.test_event(value + " hover enter");
-      const new_list = [];
-      for (let i = 0; i < this.stroke_width.length; i++) new_list[i] = this.stroke_width[i];
-      this.stroke_width.splice(0, this.stroke_width.length);
-      switch (value) {
```

Comment:
Can't believe these massive switch-case statements were in here, thanks for cleaning this up